### ISSUE TITLE
GitHub Pages does not support branches other than master & gh-pages

### ISSUE THREAD

**Issue by jeffbyrnes** [mheap/github-default-branch#16]
Note from @mheap: We can now fix this! 

We're looking for someone to add a transform that updates the repo to use the `$new` branch rather than the `$old` branch as the source for `gh-pages`. More information in https://github.com/mheap/github-default-branch/issues/16#issuecomment-686694753

### Original issue content

Well, I broke a bunch of websites today b/c when you change the name of master to anything else, and its being used by GitHub Pages, it turns off GH Pages & won’t let you turn it back on without some finagling.

So there should probably be some sort of protection for that. Not sure I’ll have time to hack something together myself, but hopefully this serves to warn others.

**Comment by mheap** [comment]
Do you know what the steps required to enable GH Pages again are?

What course of action would you recommend? Detect `gh-pages` branches and skip? Can we auto-configure gh-pages for `main`?

**Comment by jeffbyrnes** [comment]
Yep! I had to repeat them for all my repos that use GH Pages 😂 

You can do one of these to enable them:

* Push a branch called `gh-pages`
* Go to Settings & set the `master` branch to be used for GH Pages
* Go to Settings & set a directory called `docs` on the `master` branch to be used for GH Pages

We can definitely look for a `gh-pages` branch, and those are probably totally safe to convert `main` → `master` (unless `gh-pages` is the only branch, then we should skip).

However, the case of `master` being used for GH Pages is trickier. Maybe there’s a way to read the Settings of a repo, and detect that GH Pages is enabled for master in one of the two available ways? You cannot use any name besides `master` or `gh-pages` for building GH Pages from a repo.

**Comment by mheap** [comment]
This should be resolved with https://github.blog/changelog/2020-07-31-build-and-deploy-github-pages-from-any-branch-beta/

I'll leave this open until the README is updated

**Comment by elrayle** [comment]
This is now available.  Reference: [GitHub Docs - Repositories - Pages](https://docs.github.com/en/rest/reference/repos#pages)

### Via UI

* Settings -> Options tab ->
* scroll down to GitHub Pages section
* under Source, select Branch: main and in my case Folder: /docs
* Save

### Via octokit

#### Get existing settings

```
await octokit.request('GET /repos/{owner}/{repo}/pages', {
  owner: 'octocat',
  repo: 'hello-world'
})
```

returns
```
{
  "url": "https://api.github.com/repos/github/developer.github.com/pages",
  "status": "built",
  "cname": "developer.github.com",
  "custom_404": false,
  "html_url": "https://developer.github.com",
  "source": {
    "branch": "master",
    "directory": "/"
  }
}
```

From which you can grab the existing branch and directory info.

Then update with
```
await octokit.request('PUT /repos/{owner}/{repo}/pages', {
  owner: 'octocat',
  repo: 'hello-world',
  source: {
    branch: 'main',
    path: '_PATH_FROM_GET_'
  }
})
```

**Comment by mheap** [comment]
This was just released as `v2.1.0` 🎉
